azure-quantum/azure/quantum/optimization/online_problem.py

```diff
@@ -0,0 +1,22 @@
+import logging
+from typing import TYPE_CHECKING, Union, Dict
+from azure.quantum.optimization import Problem
+
+logger = logging.getLogger(__name__)
+
+__all__ = ["OnlineProblem"]
+
+
+class OnlineProblem(object):
+    def __init__(
+        self, name: str,
+        blob_uri: str,
+        **kw
+        ):
+        super(OnlineProblem, self).__init__(**kw)
+        self.name = name
+        self.uploaded_blob_uri = blob_uri
+
+    def download(self) -> Problem:
+        logger.warning("The problem will be downloaded to the client")
+        return Problem.download(self)
```

Comment:
I just realized - would this actually work with the "self" parameter? Since these are 2 different classes, and Problem.download technically doesn't take in a parameter? 

I think download_blob(self.uploaded_blob_uri) and Problem.deserialize(contents, self.name) might need to be called at this level

Reply:
The way I understood it was self is the object and calling Problem.download as a class method can enable taking the object as parameter. But actually let me test again.

Reply:
Hi, I updated the download function to handle authetication correctly. This still works with the onlineproblem when Problem.download is called as a class method with the online problem object passed to self.